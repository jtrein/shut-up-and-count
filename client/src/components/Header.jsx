import React from 'react';

import styles from '../styles/Header.styles';
import logo from '../img/logo.svg';

const Header = () => (
  <header>
    <img style={styles.logo} src={logo} alt="logo" />
    <h1 style={styles.hide}>Shut Up and Count</h1>
  </header>
);

export default Header;
