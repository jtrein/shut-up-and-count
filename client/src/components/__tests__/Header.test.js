/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

import Header from '../Header';

Enzyme.configure({ adapter: new Adapter() });

test('renders without crashing', () => {
  const div = document.createElement('div');
  render(<Header onDropAccepted={() => {}} />, div);
  unmountComponentAtNode(div);
});

test('renders expected initial components', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('header').length).toBe(1);
  expect(wrapper.find('img').length).toBe(1);
  expect(wrapper.find('h1').length).toBe(1);
});

test('h1 renders expected text', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('h1').text()).toBe('Shut Up and Count');
});

test('Snapshot: renders as expected', () => {
  const wrapper = shallow(<Header onClick={() => {}} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
