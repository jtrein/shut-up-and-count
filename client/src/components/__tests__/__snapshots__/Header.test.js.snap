// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: renders as expected 1`] = `
<header>
  <img
    alt="logo"
    src="logo.svg"
    style={
      Object {
        "maxWidth": "100%",
        "width": 250,
      }
    }
  />
  <h1
    style={
      Object {
        "clip": "rect(1px, 1px, 1px, 1px)",
        "height": "1px",
        "overflow": "hidden",
        "position": "absolute",
        "width": "1px",
      }
    }
  >
    Shut Up and Count
  </h1>
</header>
`;
